traveler_ui_kit/src/main/java/com/guestlogix/traveleruikit/fragments/CatalogItemDetailsFragment.java

```diff
@@ -140,7 +148,57 @@ public void onClick(View v) {
             }
         });
 
+        wishListToggleImageButton = view.findViewById(R.id.imagebutton_wishlist_toggle);
+        wishListToggleImageButton.setSelected(catalogItemDetails.isWishlisted());
+        wishListToggleImageButton.setOnClickListener(new View.OnClickListener() {
+            @Override
+            public void onClick(View v) {
+                List<Product> products= new ArrayList<>();
+                products.add(catalogItemDetails);
+                if (wishListToggleImageButton.isSelected()) {
+                    WishlistResult immediateResult = Traveler.wishlistRemove(
+                            products, wishlistResult,
+                            CatalogItemDetailsFragment.this);
+                    applyWishlistResult(immediateResult);
+
+                } else {
+                    wishListToggleImageButton.setSelected(true);
+                    Traveler.wishlistAdd(products, CatalogItemDetailsFragment.this);
+                }
+            }
+        });
         return view;
     }
 
+    @Override
+    public void onWishlistAddSuccess(List<CatalogItem> products) {
+        // no-op
+    }
+
+    @Override
+    public void onWishlistAddError(Error error) {
+        wishListToggleImageButton.setSelected(false);
+    }
+
+    @Override
+    public void onWishlistRemoveSuccess() {
+        // no-op
+    }
+
+    public void onWishlistRemoveError(Error error, WishlistResult result) {
+        applyWishlistResult(result);
```

Comment:
why is the SDK even returning results???

this is not how the requriements for Core SDK were

Reply:
As part of this resolution I expect to see Core updated to fix any issues related to this

Reply:
updated to match iOS new implementation